### ISSUE TITLE
Update needed for the code from tensorflow 1.4.0 to tensorflow 2

### ISSUE THREAD

**Issue by SagarDollin** [davidtellez/contrastive-predictive-coding#16]
Hi @davidtellez,
This is a great project to get started with CPC; however, while running this project in my local environment, I found that while most of the code is compatible, some lines of code are not very compatible with tensorflow 2. For example, `fit_genererator` throws a [warning](https://stackoverflow.com/questions/59380430/how-to-use-model-fit-which-supports-generators-after-fit-generator-deprecation). Also, in the versions after tensorflow 2.1.0, it has been [deprecated](https://github.com/rstudio/keras/issues/1104). 

**My proposal**
I intend to do the following things to update the project for users using tensorflow 2.
    1. Make the existing project compatible with tensorflow 2
    2. The `callbacks ` function can incentivize time to train the model and therefore try to look for an alternative, and I will also compare the results when we do not use `callback.`
    3. The project uses `LeakyRelu `as the activation in many layers; since the loss function is binary_crossentropy, I propose that we use `ReLU `instead. 

_Please let me know if you are open to the suggestions made. Also, your feedback is appreciated. Once we are on board, I will be sharing the issues that I discussed in detail. I will provide detailed information about the challenges of running the tensorflow 1.4.0 project on the tensorflow 2 environments._

Question: If this task is assigned to me, should I create a PR for the master branch or create a new branch for the updates.

Note: I'm currently focussing on the [train_mode.py](https://github.com/davidtellez/contrastive-predictive-coding/blob/master/train_model.py) file.

**Comment by 943fansi** [comment]
Hi, @SagarDollin , I am also interested with this project, have you finished this work with `tensorflow 2`?
